### DISCUSSION TITLE
Wired connection with ECU-R (SunSpec logo/ECU-ID starting with 2162xxxxxxxx)

### DISCUSSION THREAD

**Discussion by fsutter** [ksheumaker/homeassistant-apsystems_ecur#243] [Q&A]
Hi,

I have a ECU-R (SunSpec logo/ECU-ID starting with 2162xxxxxxxx) and where it is located I only have a wired connection, wifi is too far away.
In the Prerequisites section, it's written that the plugin will only work over wifi for that model.
Is it definite? If so, could someone explain me why it cannot work over a wired connection.

Many thanks in advance and kind regards,

fsutter

**Comment by jhdebruin** [comment]
Hi, can not explain why the wired doesn't work, but i can confirm that only the Wifi will work, tried every possible route with the LAN port, but no succes (as described...)

**Comment by fsutter** [comment]
Ok, so the only solution would be to replace my ECU-R with an ECU-C?

regards,

**Comment by HAEdwin** [comment]
We use an undocumented feature of the ECUs. In practice, port 8899 used, simply turned out not to be open in the Ethernet configuration, we don't know why - it is not documented. We are just enthusiastic hobbyists who want to make data available in Home Assistant, we do not represent APSystems. There is a risk that APSystems will remove this feature in the future. A working alternative would then be offered by Modbus for all models. It is definite until APSystems decides otherwise.

**Reply by fsutter** [reply]
Thanks for the explanation.

I use Modbus to "discuss" with my alarm panel.
How does Modbus work with ECUs?

Thanks

fsutter
